projects/gloo/cli/pkg/cmd/gatewayapi/convert/write.go

```diff
@@ -0,0 +1,321 @@
+				Name: namespace,
+			},
+		}
+
+		yml, err := yaml.Marshal(ns)
+		if err != nil {
+			return err
+		}
+		_, err = f.WriteString("\n---\n" + string(yml))
+		if err != nil {
+			return err
+		}
+	}
+	fmt.Printf("Generated %s/namespaces.yaml \n", dir)
+	return nil
+}
+
+func writeObjectToFile(opts *Options, wrapper domain.Wrapper, stringBytes []byte) error {
+	splitFilesByNamespace := !opts.RetainFolderStructure
+	var err error
+	if splitFilesByNamespace {
+		_, err = createSubDir(opts.OutputDir, wrapper.GetNamespace())
+		if err != nil {
+			return err
+		}
+	}
+	if opts.RetainFolderStructure {
+		// retain original file name
+		if err := appendToFile(opts.OutputDir, wrapper.GetOriginalFileName(), stringBytes); err != nil {
+			return err
+		}
+	} else {
+		// by file per namespace
+		fileName := fmt.Sprintf("%s-%s.yaml", wrapper.GetObjectKind().GroupVersionKind().Kind, wrapper.GetName())
+		if err := appendToFile(filepath.Join(opts.OutputDir, wrapper.GetNamespace()), fileName, stringBytes); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func appendToFile(outputDir string, filename string, yml []byte) error {
+	filePath := filepath.Join(outputDir, filename)
+	if err := os.MkdirAll(filepath.Dir(filePath), os.ModePerm); err != nil {
+		return err
+	}
+	f, err := os.OpenFile(filePath, os.O_APPEND|os.O_WRONLY|os.O_CREATE, 0644)
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+
+	if _, err = f.WriteString(fmt.Sprintf("\n---\n%s", removeNullYamlFields(yml))); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func createSubDir(outputDir string, subDir string) (string, error) {
+	folder := filepath.Join(outputDir, subDir)
+	err := os.MkdirAll(folder, os.ModePerm)
+	if err != nil {
+		return "", err
+	}
+	return folder, nil
+}
+
+func removeNullYamlFields(yamlData []byte) string {
+	stringData := strings.ReplaceAll(string(yamlData), "  creationTimestamp: null\n", "")
+	stringData = strings.ReplaceAll(stringData, "status:\n", "")
+	stringData = strings.ReplaceAll(stringData, "parents: null\n", "")
+	stringData = strings.ReplaceAll(stringData, "status: {}\n", "")
+	stringData = strings.ReplaceAll(stringData, "\n\n\n", "\n")
+	stringData = strings.ReplaceAll(stringData, "\n\n", "\n")
+	stringData = strings.ReplaceAll(stringData, "spec: {}\n", "")
+	stringData = strings.ReplaceAll(stringData, "    kubectl.kubernetes.io/last-applied-configuration: |\n", "")
+	stringData = strings.ReplaceAll(stringData, "  listeners: null\n", "")
+	var re = regexp.MustCompile(`\n      \{"apiVersion":.*`)
+	stringData = re.ReplaceAllString(stringData, "")
+	re = regexp.MustCompile(`\n  resourceVersion: .*`)
+	stringData = re.ReplaceAllString(stringData, "")
+	re = regexp.MustCompile(`\n  uid: .*`)
+	stringData = re.ReplaceAllString(stringData, "")
+	re = regexp.MustCompile(`\n  creationTimestamp: .*`)
+	stringData = re.ReplaceAllString(stringData, "")
+	re = regexp.MustCompile(`\n  generation: .*`)
+	stringData = re.ReplaceAllString(stringData, "")
```

Comment:
Maybe not in this section of the code but would warnings in status be cleared as well ?

Reply:
Statuses are cleared out on the loading of the objects from file because they dont differ from the snapshot than the spec so no objects should be written with any statuses i dont think. if i missed some then yeah we should do that